src/chat.rs

```diff
@@ -1901,6 +1917,66 @@ pub fn shall_attach_selfavatar(context: &Context, chat_id: ChatId) -> Result<boo
     Ok(needs_attach)
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum MuteDuration {
+    NotMuted,
+    Forever,
+    MutedUntilTimestamp(i64),
```

Comment:
should this not be u64?  what does a negative time mean here?
But this leads to other questions, why is timestamp encoded into the name?  that's not very rust-like, the type should be obvious from the type.  so `MutedDuration::Until(std::time::SystemTime)` would read a lot nicer (and matches the `MutedDuration::Forever` naming.  (I also checked out the chrono lib which we already use but I think SystemTime is the best thing here)

Reply:
negative time is used for testing, but we can also either sleep or modify the database while testing.

`std::time::SystemTime` how do I get the number representation of this type?
And how can I create it with a specific time (when creating it or loading it from the db)?

Reply:
I've formulated my comment in the form of a pull request: https://github.com/deltachat/deltachat-core-rust/pull/1250
In the mean time you've made some changes here, but I'm sure we can resolve that.

Reply:
I've rebased so you can now merge it into this PR again.

Turns out negative time is not used in testing.  The tests just use time in the past, but not times from before the epoch.  The code in the PR does all the error handling to cater for those cases (thanks to more specific types)

Reply:
thx @flub for the pr.